ocs_ci/framework/main.py

```diff
@@ -49,18 +65,20 @@ def init_ocsci_conf(arguments=None):
     parser.add_argument(
         '--ocs-version', action='store', choices=['4.2', '4.3']
     )
+    parser.add_argument('--ocs-registry-image')
     args, unknown = parser.parse_known_args(args=arguments)
-    if args.ocs_version:
+    ocs_version = args.ocs_version
+    load_config(args.ocsci_conf)
+    ocs_registry_image = framework.config.DEPLOYMENT.get('ocs_registry_image')
+    if args.ocs_registry_image:
+        ocs_registry_image = args.ocs_registry_image
+    if ocs_registry_image:
+        ocs_version = utils.get_ocs_version_from_tag(ocs_registry_image)
+    if ocs_version:
         version_config_file = os.path.join(
-            CONF_DIR, 'ocs_version', f'ocs-{args.ocs_version}.yaml'
+            CONF_DIR, 'ocs_version', f'ocs-{ocs_version}.yaml'
         )
-        args.ocsci_conf.insert(0, version_config_file)
-    for config_file in args.ocsci_conf:
-        with open(
-            os.path.abspath(os.path.expanduser(config_file))
-        ) as file_stream:
-            custom_config_data = yaml.safe_load(file_stream)
-            framework.config.update(custom_config_data)
+        load_config([version_config_file])
```

Comment:
Would be possible in principle, that loading this config file overwrites the previously parsed `ocs_registry_image` or calculated/parsed `ocs_version` value? 

In this case, we should reconcile the version again..

Reply:
(or prevent this in order to be looping...)

Reply:
This will not be overwritten as we do not have `ocs_registry_image` in any of the config file.  Only if user will pass this one this has precedence over default value which is defined in the config files like  `default_ocs_registry_image`

Reply:
Not having it in any example file is one thing. But we *could* have it in a config file, right? So in theory it could have an effect...